execution.py

```diff
@@ -42,6 +66,14 @@ def get_input_data(inputs, class_def, unique_id, outputs={}, prompt={}, extra_da
     return input_data_all
 
 def map_node_over_list(obj, input_data_all, func, allow_interrupt=False):
+    #Ensure image inputs are in BHW3 format
+    input_types = obj.INPUT_TYPES()
+    for _, v in input_types.items():
+        if isinstance(v, dict):
+            for k2, v2 in v.items():
+                if v2[0] == "IMAGE":
+                    input_data_all[k2] = [force_bhw3(x) for x in input_data_all[k2]]
```

Comment:
I think we should only validate things coming out of nodes. `map_node_over_list` is used for some cases that aren't actual node execution and there may be awkward side-effects. Additionally, if we're going to throw a warning/error, that warning/error would be attributed to a node that isn't the real culprit.

As long as we're validating all outputs, the only additional thing this protects people from is a node with output using the silly `__ne__` trick. In that case, we can't even be sure that it was intended to be an image -- it might have been an audio clip.

Reply:
These are all evaluated on execution, so you might be right, or it might work, if the variable number of nodes is codified at execution time, it shouldn't be an issue. All the information one what is being put in is gleaned from the states run through execution.py, and I made sure not to touch the get_input_data function for this reason. It needs rigorous testing with all the different kind of node presentations it can run up against. However since it mainly parses data once its already in a state where its linked to other node outputs/inputs I think it will work, unless there is something I am not understanding about how there variable input nodes work, which is possible.

Reply:
> These are all evaluated on execution, so you might be right, or it might work, if the variable number of nodes is codified at execution time, it shouldn't be an issue.

To be clear, this is definitely evaluated on execution, but this function is also called at other times -- like calling `IS_CHANGED`. In that function, it's totally legitimate to have an `IMAGE` with a value of `None` (in fact, that's the expected value since only constant inputs are available when `IS_CHANGED` is called before the graph has begun execution.

I would put the validation right after the call to `get_output_data` in `recursive_execute`. At that point, you have information about the node that was executed like its output types.